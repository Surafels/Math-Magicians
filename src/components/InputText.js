import React from 'react';
import PropTypes from 'prop-types';

function InputText(props) {
  const { value, className, onClick } = props;

  return (
    <input
      className={className}
      type="button"
      value={value}
      onClick={() => onClick(value)} // Pass the button value to the click handler
    />
  );
}

InputText.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};
InputText.defaultProps = {
  className: null,
};

export default InputText;
