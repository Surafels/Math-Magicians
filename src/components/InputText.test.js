import React from 'react';
import renderer from 'react-test-renderer';
import InputText
 from './InputText';

test('Should render input correctly', () =>{
const onClick = jest.fn();

const tree = renderer.create(
    <InputText value="Test" className="custom-class" onClick ={onClick} />) .toJSON();

expect(tree).toMatchSnapshot();
});