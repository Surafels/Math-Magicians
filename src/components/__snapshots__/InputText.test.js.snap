// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render input correctly 1`] = `
<input
  className="custom-class"
  onClick={[Function]}
  type="button"
  value="Test"
/>
`;
